Raysu 2023 0519

ruihengsu opened the pull request:
**Open connections to the lakeshore magnet power supply parameters corrected**. Tested to work in the LD system
Errors associated with connecting to the IPS power supply were also corrected.

ruihengsu committed:
some small changes, need to revise before merging

ruihengsu committed:
Update

Update

ruihengsu committed:
Smaller changes

good

sillyPhotons commented:
The most important changes are the ipfs for the LS625 and IPS120 power supplies!

johann997 commented on Required/Scans.ipf:
```diff
@@ -696,8 +697,11 @@ function Scank2400(instrID, startx, finx, channelsx, numptsx, delayx, rampratex,
+			rampK2400Voltage(S.instrIDx, setpointx, ramprate=S.rampratex)
+		else
+		   setK2400Voltage(S.instrIDx, setpointx)
+		endif
```
Not essential, but nice to keep orderly indenting.

johann997 commented on Required/Scans.ipf:
```diff
@@ -1086,11 +1089,18 @@ function ScanMultipleK2400LS625Magnet2D(keithleyIDs, start, fin, numptsx, delayx
+		fast=0
+	endif
+	
+	
```
In IGOR if the parameter is default it is already set to 0. But it is good to be explicit. 
A nice 1 liner would be:
fast = paramisdefault(fast) ? 1 : fast

johann997 commented on Required/Scans.ipf:
```diff
@@ -1121,8 +1134,13 @@ function ScanLS625Magnet(instrID, startx, finx, numptsx, delayx, [y_label, comme
-		sc_sleep(S.delayx)
+		if(fast==1)
+			setlS625field(S.instrIDx, setpointx) 
+			sc_sleep(max(S.delayx, (S.delayx+60*abs(finx-startx)/numptsx/ramprate)))
```
What does this math do? Looks like you have the time in minutes for a ramp to complete divided by the number of points in x. So if you take 1 data point this is the same as setlS625fieldwait(S.instrIDx, setpointx) but if you take 1e6 data points then you will start collecting data immediately??

johann997 commented on Required/srs830.ipf:
```diff
@@ -247,6 +247,18 @@ function AutoSRSPhase(instrID) // Units: deg
 	writeInstr(instrID, "APHS \n")
 end
 
+function AutoSRSSens(instrID) 
```
Write a comment on what this function does.

johann997 commented on Scan_n_D_Bxyz.ipf:
```diff
@@ -324,6 +324,99 @@ function Scan_n(instrIDx,instrIDy,fixedD,startn,finn,numptsn,delayn,rampraten, [
 
 
+
+function Scan_n_limit(instrIDx,instrIDy,fixedD,startn,finn,numptsn,delayn,rampraten, limitx, limity, [y_label, comments, nosave]) //Units: mV
```
What does this function do? Write a comment.

johann997 committed:
Merge branch 'master' into raysu_2023_0519

johann997 commented:
Everything looks fine to me. Most of the changes were in Scan_n_D_Bxyz.ipf which I never use. 
In general, it would be nice to see more comments, especially when you add a new function or new parameters to a function. But overall seems fine to me.

FolkLab committed:
New updates to the scan_n_d codes incorporating a division ratio in the gate capacitance

New updates to the scan_n_d codes incorporating a division ratio in the gate capacitance

FolkLab committed:
after last experiment
